checks.d/file.py

```diff
@@ -49,28 +49,39 @@ def check(self, instance):
 
         path = instance['path']
         expect = instance['expect']
+        min_age = instance.get('present_minimum_age_seconds', 0)
 
         status, statinfo = self.stat_file(path)
-
         tags = [
             'expected_status:' + expect,
             'path:' + path
         ]
 
+        # Set a default
+        file_age = -1
+        m_age = -1
+        timestamp = time.time()
+        if status == self.STATUS_PRESENT:
+            timestamp = statinfo.st_ctime
+            # Set this for the metric later
+            file_age = time.time() - timestamp
+            # We use the mtime because we cant change the ctime for testing.
+            m_age = time.time() - statinfo.st_mtime
+
         # Emit a service check:
         msg = "File %s is %s" % (path, expect)
         check_status = AgentCheck.OK
         if status != expect:
-            check_status = AgentCheck.CRITICAL
-            msg = "File %s that was expected to be %s is %s instead" % (path, expect, status)
+            if (status == self.STATUS_PRESENT and m_age > min_age) or expect == self.STATUS_PRESENT:
```

Comment:
I think the `>` is flipped here?

Reply:
Hm, I think I got confused. Is the idea to not turn critical if the file is younger than the configured minimum age?

Would probably be better/more understandable to do a separate check above this one for `min_age > m_age` and break out early if so.

Reply:
Separately, I don't like `m_age` much as a name... to similar to `min_age`?